javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -218,7 +219,7 @@ public final Node setComment(final Comment comment) {
     }
 
     public boolean hasJavaDocComment() {
-        return hasComment() && getComment() instanceof JavadocComment;
+        return hasComment() && getComment().get() instanceof JavadocComment;
```

Comment:
hasComment() actually does a check to see if the comment exists, so no worries here :)